tests/apply_tests/test_models.py

```diff
@@ -0,0 +1,88 @@
+from unittest.mock import Mock
+
+from apply.constants import ApplicationStatus
+from apply.models.application import Application
+
+
+def test_match_forms_to_state(apply_test_client):
+    application = Application(
+        id="app123",
+        reference="ref123",
+        account_id="acc123",
+        status=ApplicationStatus.IN_PROGRESS.name,
+        fund_id="fund456",
+        round_id="round789",
+        project_name="Test Project",
+        date_submitted=None,
+        started_at=None,
+        last_edited=None,
+        language="en",
+        forms=[
+            {"name": "form_1", "status": ApplicationStatus.COMPLETED.name},
+            {"name": "form_2", "status": ApplicationStatus.NOT_STARTED.name},
```

Comment:
these should be `Form` instances according to type hints

Reply:
Typehint seems incorrect..

```python
        # fill the section/forms with form state from the application
        for form_state in self.forms:
            # find matching form in sections
            for section in sections_config:
                for form_in_config in section["forms"]:
                    if form_in_config["form_name"] == form_state["name"]:
                        form_in_config["state"] = form_state
```

if `self.forms` is indeed a `List[Form]` then we'd us `form_state.name` rather than `form_state["name"]`

Reply:
you're right, the code is wrong and should feel bad 👍